src/components/PopoverMenu/BasePopoverMenu.js

```diff
@@ -26,13 +26,88 @@ const defaultProps = {
 };
 
 class BasePopoverMenu extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            activeMenuIndex: -1,
+        };
+        this.onModalHide = this.onModalHide.bind(this);
+    }
+
+    componentDidMount() {
+        if (!this.props.allowKeyboardNavigation) {
+            return;
+        }
+        this.setupEventHandlers();
+    }
+
+    componentWillUnmount() {
+        if (!this.props.allowKeyboardNavigation) {
+            return;
+        }
+        this.cleanupEventHandlers();
+    }
+
+    onModalHide() {
+        this.setState({activeMenuIndex: -1});
+        this.props.onMenuHide();
+    }
+
+    setupEventHandlers() {
+        if (!document) {
+            return;
+        }
+        this.keyDownHandler = (keyBoardEvent) => {
+            if (keyBoardEvent.key.startsWith('Arrow')) {
+                this.highlightActiveMenu(keyBoardEvent.key);
+                return;
+            }
+
+            if (keyBoardEvent.key === 'Enter' && this.state.activeMenuIndex !== -1) {
+                this.props.onItemSelected(this.props.menuItems[this.state.activeMenuIndex]);
+            }
+        };
+        document.addEventListener('keydown', this.keyDownHandler, true);
```

Comment:
Use KeyboardShorcut lib here.

Reply:
Umm a little confused, how `KeyboardShortcut` lib will work here? I thought `KeyboardShortcuts` is to bind a shortcut to the action and subscribe, and here we need for navigation. Can you throw some light?

Reply:
Of yeah Sorry This is fine..

Reply:
I'm confused why we couldn't or wouldn't use `KeyboardShortcut` for this PR...